products/bigquery/terraform.yaml

```diff
@@ -0,0 +1,63 @@
+# Copyright 2019 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Provider::Terraform::Config
+legacy_name: 'bigquery'
+overrides: !ruby/object:Overrides::ResourceOverrides
+  Dataset: !ruby/object:Overrides::Terraform::ResourceOverride
+    id_format: "{{project}}:{{dataset_id}}"
+    import_format: ["{{project}}:{{dataset_id}}", "{{project}}/{{dataset_id}}", "{{dataset_id}}"]
+    examples:
+      - !ruby/object:Provider::Terraform::Examples
+        name: "bigquery_dataset_basic"
+        primary_resource_id: "dataset"
+        vars:
+          dataset_id: "example_dataset"
+    properties:
+      access: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
+      defaultTableExpirationMs: !ruby/object:Overrides::Terraform::PropertyOverride
+        validation: !ruby/object:Provider::Terraform::Validation
+          function: 'validateDefaultTableExpirationMs'
+      datasetReference: !ruby/object:Overrides::Terraform::PropertyOverride
+        exclude: true
+      id: !ruby/object:Overrides::Terraform::PropertyOverride
+        exclude: true
+      location: !ruby/object:Overrides::Terraform::PropertyOverride
+        input: true
+        custom_flatten: templates/terraform/custom_flatten/bigquery_dataset_location.go.erb
+      name: !ruby/object:Overrides::Terraform::PropertyOverride
+        exclude: true
+    custom_code: !ruby/object:Provider::Terraform::CustomCode
+      constants: templates/terraform/constants/bigquery_dataset.go.erb
+      custom_import: templates/terraform/custom_import/bigquery_dataset.go.erb
```

Comment:
Given that we have `import_format`, is the custom import still needed? If so, can you add a comment explaining why?

Reply:
Needed because auto importers don't work with : or / as separators. Added a note on the code and in terraform.yaml on what is the `import_format` for.

Reply:
I don't think that's true, it should work with whatever separator you specify. There was recently a fix made to ensure the order of the import ids are from most to least specific, which may have been the problem you were seeing. Can you rebase and see if that fixes the import problems you had before?

Reply:
Indeed, got rid of the custom_import after rebasing. Changes pushed.